src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -130,22 +128,26 @@ export class TechRecordSummaryComponent implements OnInit {
     this.store.dispatch(updateEditingTechRecord({ techRecord: this.vehicleTechRecordCalculated }));
   }
 
-  findAxleToRemove(axles: Axle[]): number {
-    let previousAxleRow = 1;
+  handleFormState(event: any): void {
+    this.vehicleTechRecordCalculated = merge(cloneDeep(this.vehicleTechRecordCalculated), event);
 
-    for (const ax of axles) {
-      if (ax.axleNumber === previousAxleRow) {
-        previousAxleRow += 1;
-      } else {
-        return ax.axleNumber! - 1;
-      }
+    if (event.axles && event.axles.length < this.vehicleTechRecordCalculated.axles.length) {
```

Comment:
Do you need the truthy check on `event.axles`, is it not possible to just do:
`if (event.axles?.length < this.vehicleTechRecordCalculated.axles.length)`

Reply:
The first `event.axles` check is explicit. It checks whether the event was triggered by the axles section.

Reply:
This is the exact one that I was saying broke so if it works with this I will quit